### DISCUSSION TITLE
Detecting empty message produced by "SQL no rows"

### DISCUSSION THREAD

**Discussion by pinebit** [redpanda-data/connect#1811] [General]
Hello!

I have sql_select => unarchive flow, and if my SQL query returns no rows by design, then I am getting a failure for "unarchive" task, because of empty message.
How shall I handle this case properly? I usually want to execute a fallback processor that would be run only if the query returned no rows.
Also, I would like that my "unarchive" task do not yield any errors. That is, I probably need to be able to test for empty message before even executing "unarchive" task. But I don't want to just drop/delete that empty message and instead execute another processor, such as "metric" to count such events.

Thank you!

**Comment by mihaitodor** [comment]
Hey @pinebit! I think you can leverage the [`switch` processor](https://www.benthos.dev/docs/components/processors/switch) for that purpose. The [examples section](https://www.benthos.dev/docs/components/processors/switch#examples) should help you see how to use it, but you can basically leverage the bloblang [`length()` method](https://www.benthos.dev/docs/guides/bloblang/methods#length-1) in the `check` field to determine if you have an empty message and then execute your `metric` processor under it.

**Reply by pinebit** [reply]
Thank you @mihaitodor

**Reply by pinebit** [reply]
I ended up doing this:
```yaml
    - switch:
        - check: (this | "").length() == 0
```
without coalescing `this.length()` yields error:
> Failed to test case 0: failed assignment (line 1): unable to reference message as structured (with 'this'): message is empty

Is this correct?

**Reply by mihaitodor** [reply]
Yep, that works. However, you can use `content()` instead of `(this | "")`, since that will be cleaner (and faster)
